### ISSUE TITLE
Improvements to the FLAG_FAIL_FAST option

### ISSUE THREAD

**Issue by hellais** [ooni/pipeline#185]
Currently the centrifugation script has a hardcoded constant called `FLAG_FAIL_FAST` which defaults to being set to `False`, but when it's set to true it will propagate errors.

As seen here: https://github.com/ooni/pipeline/pull/177#discussion_r279317109, the fact that exceptions are not obviously thrown can lead to bugs creeping up.

Moreover it's probably beneficial to at least run the CI tests with this flag on.

I think at bare minimum it should be possible to enable this flag as an argument to `centrifugation`, but maybe we could also revisit this pattern.

Here are some comments from @darkk which may be of help to inform a decision:

> Yes, garbage-in-garbage-out.
> That's done that way so the malformed measurement is still indexed and marked with `exc` and does not block the pipeline.
> You may enable `FLAG_FAIL_FAST` to avoid exceptions suppression.

> Storing exception as 32bit hashsum may sound like a strange idea, that's done to prevent database bloat in a pathological case as 1M of traces is not something trivial to deal with.
> That idea is to use the hashsum to triage "failures" and pay attention to "popular" failures first.

> FTR, the hash is computed by [exc_hash()](https://github.com/ooni/pipeline/blob/065cccdfeb531e93a22d2aacc05ec05e990f99ee/af/shovel/centrifugation.py#L996-L1006).

**Comment by hellais** [comment]
Note: I tried enabling `FLAG_FAIL_FAST` on the CI which operates on the bucket `2018-05-07`, yet it fails due to:

```
  File "/mnt/af/shovel/centrifugation.py", line 1803, in main
    meta_pg(opt.autoclaved_root, bucket, opt.postgres)
  File "/mnt/af/shovel/centrifugation.py", line 1756, in meta_pg
    copy_data_from_autoclaved(pgconn, stconn, in_root, bucket, bucket_code_ver)
  File "/mnt/af/shovel/centrifugation.py", line 632, in copy_data_from_autoclaved
    row = feeder.row(msm_no, datum)
  File "/mnt/af/shovel/centrifugation.py", line 1511, in row
    ret += self._row(msm_no, r)
  File "/mnt/af/shovel/centrifugation.py", line 1525, in _row
    headers = ujson.dumps(http_headers(headers))
  File "/mnt/af/shovel/centrifugation.py", line 139, in http_headers
    return {pg_uniquote(k): pg_uniquote(v) for k, v in headers.iteritems()}
  File "/mnt/af/shovel/centrifugation.py", line 139, in <dictcomp>
    return {pg_uniquote(k): pg_uniquote(v) for k, v in headers.iteritems()}
  File "/mnt/af/shovel/oonipl/pg.py", line 184, in pg_uniquote
    assert isinstance(s, unicode)
AssertionError
```

**Comment by darkk** [comment]
Yep, that's clear indicator of the lack of the ongoing _cleanup_ process implied by #137.
